DS13/code/modules/overmap/warpcore.dm

```diff
@@ -0,0 +1,676 @@
+			alignment = max_alignment
+		else
+			to_chat(user, "<span class='warning'>Realignment failed continued failure risks damage to dilithium crystal sample. Rotating command sequence.</span>")
+			playsound(src, 'DS13/sound/effects/warpcore/overload.ogg', 100, 1)
+			combo_target = "[pick(letters)][pick(letters)][pick(letters)][pick(letters)][pick(letters)]"
+			absorb_damage(5) //Penalty for fucking it up. You risk destroying the crystal...
+		combocount = 0
+		combo = null
+
+//HOW TO RATIO:
+//LITTLE NUMBER / BIG NUMBER:
+// 15/30 ->  0.5
+/*
+
+The antimatter | matter ratio is preset and constant, if it's powered. It gives a preset amount of power. The fun comes from maintaining it.
+
+*/
+
+/datum/supply_pack/engine/dilithium_matrix
+	name = "Dilithium Matrix Crate"
+	desc = "A replacement dilithium matrix and crystal sample for repairing warp cores. Requires engineering access to open."
+	cost = 10000
+	access = ACCESS_ENGINE_EQUIP
+	contains = list(/obj/item/dilithium_matrix,/obj/item/dilithium)
+	crate_name = "Dilithium Matrix Crate"
+	crate_type = /obj/structure/closet/crate/secure/engineering
+	dangerous = TRUE
+
+/obj/machinery/power/warp_core //here's where the magic happens. This thing takes matter and antimatter. The ratio at which theyre supplied determines the power output and chance of EXPLODING
+	name = "warp core"
+	desc = "A gigantic machine that amounts to a reaction chamber. It is fitted with magnetic constrictors for the containment of highly volatile antimatter while also being linked to the ship's matter and antimatter injectors which provide a reaction substrate."
+	icon = 'DS13/icons/obj/power/warpcore/core.dmi'
+	icon_state = "core-empty"
+	pixel_x = -29
+	layer = 4
+	anchored = TRUE
+	density = TRUE
+	use_power = 0
+	obj_integrity = 300 //Super tough.
+	var/active = FALSE //Starts off. You have to power it with injectors.
+	var/toggleable = TRUE //If you leave it to breach for too long, you'll no longer be able to turn it off!
+	var/obj/machinery/atmospherics/components/binary/pump/outlet
+	var/matter_stream = FALSE
+	var/antimatter_stream = FALSE
+	var/obj/machinery/power/matter_injector/MI
+	var/obj/machinery/power/matter_injector/antimatter/AI
+	var/obj/item/dilithium_matrix/dilithium_matrix
+	var/datum/looping_sound/warp/soundloop
+	var/containment = 100 //Don't let this drop to 0
+	var/max_containment = 100
+	var/breaching = FALSE //used to avoid soundspam
+	var/ejected = FALSE //B'ELANNA, EJECT THE CORE
+	var/ejecting = FALSE
+	var/obj/structure/overmap/linked
+	var/obj/item/radio/Radio
+	var/voice_ready = TRUE //radio announcement
+	var/voice_cooldown = 300
+	var/radio_key = /obj/item/encryptionkey/headset_eng
+	var/list/coils = list() //fuck I burnt my coil cyka
+	req_access = list(ACCESS_ENGINE_EQUIP)
+	//Performance penalty vars. If you don't look after your warp core, your crystal will burn out faster (if you leave this too long, you risk a core breach)
```

Comment:
Which vars does this comment refer to?

Reply:
Planned feature that I scrapped. Best remove em